numpy/_core/src/umath/ufunc_object.c

```diff
@@ -6549,6 +6573,17 @@ static PyGetSetDef ufunc_getset[] = {
 };
 
 
+/******************************************************************************
+ ***                          UFUNC MEMBERS                                 ***
+ *****************************************************************************/
+
+static PyMemberDef ufunc_members[] = {
+    {"__dict__", T_OBJECT, offsetof(PyUFuncObject, dict),
+     READONLY},
+    {NULL},
+};
```

Comment:
Do you really need this if you havea  `tp_dictoffset`?

Reply:
I took these parts from the original one: https://github.com/numpy/numpy/pull/27383.

I think `PyMemberDef ufunc_members[]` is needed. Without it my test fails with:
```
FAILED numpy/_core/tests/test_umath.py::TestSpecialMethods::test_ufunc_docstring - AttributeError: 'numpy.ufunc' object has no attribute '__dict__'
```

Reply:
Thanks, I thought it may be automatically filled in/inherited by `PyType_Ready`